mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/PreBuildTestOptimizer.java

```diff
@@ -75,8 +75,8 @@ private void configureExtension(MavenSession session, Configuration configuratio
         final MavenProjectConfigurator mavenProjectConfigurator = new MavenProjectConfigurator(configuration);
         session.getAllProjects().forEach(mavenProject -> {
             final Model model = mavenProject.getModel();
-            mavenProjectConfigurator.addRequiredDependencies(model);
             mavenProjectConfigurator.configureTestRunner(model);
+            mavenProjectConfigurator.addRequiredDependencies(model);
```

Comment:
now it is - in the first step (`configureTestRunner`) it [sets the surefireVersion ](https://github.com/arquillian/smart-testing/blob/master/mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenProjectConfigurator.java#L90) which is then read in the `addRequiredDependencies` method [here](https://github.com/arquillian/smart-testing/blob/master/mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenProjectConfigurator.java#L61).
Actually, it opens an interesting question - why do we have two methods that are called in sequence and not only one method that would handle both operations. 
I'll open a PR for it